src/makeCalculator.js

```diff
@@ -37,7 +37,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(num) {
+      calculator.result += num;
+
+      return calculator;
+    },
+    subtract(num) {
+      calculator.result -= num;
+
+      return calculator;
+    },
+    multiply(num) {
+      calculator.result *= num;
+
+      return calculator;
+    },
+    divide(num) {
+      calculator.result /= num;
+
+      return calculator;
+    },
+    reset() {
+      calculator.result = 0;
+
+      return calculator;
+    },
+    operate(callback, num) {
+      callback(num);
```

Comment:
Use 'this.operate.call(this, num)' instead of 'callback(num)' to ensure proper 'this' binding.